crypto/s2n_certificate.h

```diff
@@ -53,30 +39,45 @@ typedef enum {
     S2N_CERT_TYPE_ECDSA_FIXED_ECDH = 66,
 } s2n_cert_type;
 
-struct s2n_cert_public_key {
+struct s2n_cert {
```

Comment:
What are your thoughts on further breaking up these structs and also creating `s2n_cert_public_key` and `s2n_cert_private_key` structs?